webapp/src/app.tsx

```diff
@@ -1462,10 +1462,29 @@ export class ProjectView
                 pubCurrent: false
             }
         }
+
         return workspace.installAsync(h, project.text)
             .then(hd => this.loadHeaderAsync(hd, editorState));
     }
 
+    importTutorialAsync(md: string) {
+        try {
+            const [options, editor] = getTutorialOptions(md, "untitled", "untitled", "", false);
+            const dependencies = pxt.gallery.parsePackagesFromMarkdown(md);
+
+            return this.createProjectAsync({
+                name: "untitled",
+                tutorial: options,
+                preferredEditor: editor,
+                dependencies
+            });
+        }
+        catch (e) {
+            Util.userError("Could not import tutorial");
```

Comment:
add a tick?

Reply:
It doesn't swallow it, it rejects the promise. I'm not really sure that there is a good reason to pollute our exceptions with errors generated by the tutorial tool